# Logistic Regression

In [15]:
# yang di import
import pandas as pd
import numpy as np
import sklearn.model_selection as ms
import sklearn.linear_model as lm
import sklearn.metrics as met

In [10]:
df1 = pd.read_csv('calonpembeli.csv')
df1.describe()

ID         Usia       Status      Kelamin  Memiliki_Mobil  \
count  1000.000000  1000.000000  1000.000000  1000.000000      1000.00000   
mean    500.500000    43.532000     1.469000     0.481000         0.95200   
std     288.819436    12.672078     1.073402     0.499889         0.80146   
min       1.000000    24.000000     0.000000     0.000000         0.00000   
25%     250.750000    33.000000     1.000000     0.000000         0.00000   
50%     500.500000    43.000000     1.000000     0.000000         1.00000   
75%     750.250000    53.000000     2.000000     1.000000         2.00000   
max    1000.000000   164.000000     3.000000     1.000000         4.00000   

       Penghasilan   Beli_Mobil  
count   1000.00000  1000.000000  
mean     270.09000     0.633000  
std       95.23681     0.482228  
min       95.00000     0.000000  
25%      187.00000     0.000000  
50%      258.50000     1.000000  
75%      352.25000     1.000000  
max      490.00000     1.000000

di atas Usia maling tinggi adalah 164 tahun maka kita saring. hanya sampai 100 saja

In [11]:
df1 = df1[df1["Usia"] <= 100]

kita perlu melihat apakah ada data kosong (null) di dalamnya.

In [12]:
df1.isnull().sum()

ID                0
Usia              0
Status            0
Kelamin           0
Memiliki_Mobil    0
Penghasilan       0
Beli_Mobil        0
dtype: int64

Jika ada data yang NaN/Null maka kita Hilangkan dgn df1.dropna()
jika tidak ada maka langsung saja.

ingin melihat total pembeli sama yg tidak beli

In [13]:
df1["Beli_Mobil"].value_counts()

1    633
0    365
Name: Beli_Mobil, dtype: int64

0 adalah tidak jadi beli dan 1 adalah benar benar pembeli mobil.

Kita buat data Training dan data test. data training 80% dan data test 20% ini harus dipilh secara acak.

In [14]:
X = df1[['Usia','Status','Kelamin','Memiliki_Mobil', 'Penghasilan']]
y = df1.Beli_Mobil
X_train,X_test,y_train,y_test = ms.train_test_split(X,y, test_size = 0.2, random_state=0) 

## Model

In [16]:
model = lm.LogisticRegression(solver='lbfgs')
model.fit(X_train,y_train)

LogisticRegression()

cari tau nilai slope/koefisien (m) dan intercept (b).

In [17]:
print(model.coef_)

[[-0.01594373 -0.17093184 -1.03941657  0.07720323  0.05686709]]


In [18]:
y_prediksi = model.predict(X_test)
print(y_prediksi)

[1 0 0 1 1 1 0 0 1 1 1 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 1 1 1 1 0 1 1 1 0 1 1
 1 1 1 0 1 1 1 1 1 1 0 1 0 0 1 0 1 1 1 1 1 1 1 0 1 0 0 0 0 0 1 0 1 1 1 1 1
 1 1 1 0 1 1 1 0 1 1 1 1 1 1 1 1 1 1 1 0 0 1 0 0 1 1 1 1 1 1 0 0 1 1 1 1 0
 1 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 0 0 1 1 0 1 0 0 0 1 1 1 0 0 0 1 1
 1 1 1 1 1 1 1 0 0 1 1 1 1 1 0 1 0 0 1 1 1 1 1 0 0 1 1 1 1 1 1 0 1 1 0 1 1
 0 1 1 1 1 0 0 1 1 0 1 0 1 0 1]


In [19]:
X_test.head()

Usia  Status  Kelamin  Memiliki_Mobil  Penghasilan
272    54       3        0               0          416
727    45       2        1               1          131
31     37       2        1               1          203
344    58       0        0               1          265
606    53       1        0               1          348

In [20]:
X_test.head(1)

Usia  Status  Kelamin  Memiliki_Mobil  Penghasilan
272    54       3        0               0          416